core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -129,10 +134,19 @@ private void throwBadRequest(StackView stack) {
                 stack.getName(), stack.getStatus()));
     }
 
-    public void validateStackStatusForStartHostGroup(StackView stack) {
-        if (!(stack.isAvailable() || stack.isAvailableWithStoppedInstances())) {
+    public void validateStackStatusForStartHostGroup(StackDto stack) {
+        CmTemplateProcessor processor = cmTemplateProcessorFactory.get(stack.getBlueprint().getBlueprintText());
```

Comment:
Let's change the logic here as discussed on the call. Apologies for the late change.

We are now looking at the healthy of the hosts running the specific services.
Without any changes in place, and irrespective of <7.2.12 or >=7.2.12 - the current logic relies on all cluster nodes being in SERVICES_HEALTHY state (with the exception of some nodes being in STOPPED state).



Changes: 
1. 7.2.12 and 7.2.15 check is not required any longer.
2. Continue to get the dependent hostGroups based on the components being scaled-up or scaled-down.
3. As long as these dependent HostGroups are in SERVICES_HEALTHY state - we will allow the scaling operation.
4. (Also the fact that there's no background operation currently running continues to be a check)
5. If we cannot determine the dependent host groups - fall back to the current check (isAvailable || isAvailableWithStoppedInstances)

Rationale for not worrying about <7.2.12
* Today's scaling operations already don't check the SERVICE_HEALTH. The current change does not check the SERVICE_HEALTH either.
* Moving the checks to the dependent hostGroups makes no difference from the SERVICE_HEALTH perspective, but does help towards allowing a scaling operation if a different hostGroup is marked bad for some reason.
* Example: A DataNode marked as NODE_DELETED will now not block a scale-up. The ResourceManager health will be the same as it is in today' health check for older clusters, so is no different, and the best we can do right now - without explicitly calling multiple APIs on the CM side.